### ISSUE TITLE
deflake concurrent connection tests

### ISSUE THREAD

**Issue by shawkins** [fabric8io/kubernetes-client#4733]
On two recent runs I'm seeing issues with the 20 second wait https://github.com/fabric8io/kubernetes-client/actions/runs/3839873556/jobs/6538227145#step:4:38275 in the websocket test (one with jdk the other with okhttp).  Running locally, I've reproduced that and sometimes failed on one of the other tests as well.

One issue could be a problem with the HttpServer - if you remove the waits before returning the response, then jetty seems to consistently fail at 200 requests.  JDK seems to fail at around 400 requests.  Checking the websocket futures it's clear in this scenario that we're flooding the server to the point of failure: 

```Caused by: org.eclipse.jetty.websocket.api.exceptions.UpgradeException: org.eclipse.jetty.websocket.core.exception.UpgradeException: Failed to upgrade to websocket: Unexpected HTTP Response Status Code: 500 Internal Server Error
	at org.eclipse.jetty.websocket.common.JettyWebSocketFrameHandler.convertCause(JettyWebSocketFrameHandler.java:532)
	... 27 more
Caused by: org.eclipse.jetty.websocket.core.exception.UpgradeException: Failed to upgrade to websocket: Unexpected HTTP Response Status Code: 500 Internal Server Error
	... 21 more
java.util.concurrent.ExecutionException: io.fabric8.kubernetes.client.http.WebSocketHandshakeException
``` 

The test logic is not aware of that possibility as the websocket future is not being checked.

The strategy of introducing a wait before returning the headers also holds a thread per request.  The tight loop creation of requests for jdk in particular is prone to creating an additional thousand threads because the tasks are executed to quickly against a cached thread pool for thread reuse.  This means there can be 2 or more thousand threads running at the same time (I believe okhttp will use 2 threads per websocket as well), which could be an issue on older / smaller vms (you'll need several gigs of stack space).  My local system doesn't seem to have a thread allocation failure until around 32000 threads, so I'm not sure how much this is an issue with my local testing.  I did notice that with the header waits left in I could get more successful test completions with a wait longer than 20 seconds - sometimes up to 60 seconds - but wasn't able to quickly track down what was contributing to that additional time.

cc @manusa

**Comment by manusa** [comment]
Considering the 1 MiB per thread rule, 1024 connections might fall in the upper limit for GitHub CI's infrastructure depending on the HttpClient implementation. So yes, at some point the idea is that there are at least 2*1024 threads server-side and n*1024 threads client-side.
In my original test runs, GitHub was resilient enough to pass with these conditions, but apparently it isn't. To deflake the test we could just reduce the number of expected connections for this specific scenario, especially because the other `http1WebSocketConnectionsBeforeUpgrade` test is effectively testing the edge-case (that concerns me the most).

**Comment by shawkins** [comment]
Another occurrence https://github.com/fabric8io/kubernetes-client/actions/runs/3886155977/jobs/6630919439#step:4:9583

**Comment by shawkins** [comment]
Addressed by #4752
